cluster/examples/kubernetes/ceph/create-external-cluster-resources.py

```diff
@@ -0,0 +1,283 @@
+try:
+    ModuleNotFoundError
+except:
+    ModuleNotFoundError = ImportError
+
+try:
+    import rados
+except ModuleNotFoundError as noModErr:
+    print("Error: %s\nExiting the script..." % noModErr)
+    sys.exit(1)
+
+try:
+    # for 2.7.x
+    from StringIO import StringIO
+except ModuleNotFoundError:
+    # for 3.x
+    from io import StringIO
+
+
+class ExecutionFailureException(Exception):
+    pass
+
+
+class RadosJSON:
+    EXTERNAL_USER_NAME = "client.healthchecker"
+
+    @classmethod
+    def gen_arg_parser(cls):
+        argP = argparse.ArgumentParser()
+        argP.add_argument("--conf", "-c",
+                          help="Provide a ceph conf file.", type=str)
+        argP.add_argument("--run-as-user", "-u",
+                          help="Provides a user name to check the cluster's health status",
+                          default=cls.EXTERNAL_USER_NAME, type=str)
+        argP.add_argument("--json", "-j", action="store_true",
+                          default=False, help="Provides a JSON output")
+        argP.add_argument("--bash", "-b", action="store_true",
+                          default=False, help="Provides BASH script compatible output")
+        argP.add_argument("--output", "-o", default="",
+                          help="Output will be stored into the provided file")
+        argP.add_argument("--cephfs-filesystem-name", default="", required=True,
+                          help="Provides the name the filesystem")
+        argP.add_argument("--cephfs-data-pool-name", default="", required=True,
+                          help="Provides the name of the cephfs data pool")
+        argP.add_argument("--rdb-datapool-name", default="", required=True,
+                          help="Provides the name of the RDB datapool")
+        argP.add_argument("--namespace", default="",
+                          required=True, help="Namespace where CephCluster is running")
+        argP.add_argument("--rgw-pool-prefix", default="",
+                          help="RGW Pool prefix")
+        argP.add_argument("--rgw-realm-name", default="",
+                          help="RGW Realm name")
+        argP.add_argument("--rgw-zonegroup-name", default="",
+                          help="RGW ZoneGroup name")
+        argP.add_argument("--rgw-zone-name", default="", help="RGW Zone name")
+        argP.add_argument("--rbd-data-pool-name", default="",
+                          help="RBD data pool name")
+        argP.add_argument("--rgw-endpoint", default="", required=True,
+                          help="Rados GateWay endpoint")
+        return argP.parse_args()
+
+    def __init__(self):
+        self.out_map = {}
+        self._arg_parser = self.gen_arg_parser()
+        self.output_file = self._arg_parser.output
+        self.conf = self._arg_parser.conf
+        self.run_as_user = self._arg_parser.run_as_user
+        if not self.run_as_user:
+            self.run_as_user = self.EXTERNAL_USER_NAME
+        if self.conf:
```

Comment:
self.conf?

Reply:
`self.conf` is the ceph-config file, an optional argument needed for `rados.Rados(conf)` instance creation.
Changed the name to `ceph_conf`